packages/jest-cli/src/TestRunner.js

```diff
@@ -376,12 +385,75 @@ class TestRunner {
       this.addReporter(new CoverageReporter());
     }
 
-    this.addReporter(new SummaryReporter());
     if (config.notify) {
       this.addReporter(new NotifyReporter(this._startRun));
     }
   }
 
+  /**
+   * _setupDefaultReporters
+   * Method for setting up the default reporters
+   */
+  _setupDefaultReporters() {
+    const config = this._config;
+
+    this.addReporter(
+      config.verbose
+        ? new VerboseReporter(config)
+        : new DefaultReporter()
+    );
+
+    this.addReporter(new SummaryReporter());
+  }
+
+  /**
+   * Add Custom reporters to Jest
+   * Custom reporters can be added to Jest using the reporters option in Jest
+   * Config. The format for adding a custom reporter is following
+   *
+   * "reporters": [
+   *    ["reporterPath/packageName", { option1: 'fasklj' }], 
+   *    // Format if we want to specify options
+   *    
+   *    "reporterName"
+   *    // Format if we don't want to specify any options
+   * ]
+   */
+  _addCustomReporters(reporters: any) {
```

Comment:
yeah. this should be called from the constructor (or `setupReporters()` method).
i also think it should probably take the entire `config` as an argument, since it's aware of the configuration and it's better to encapsulate this logic in once place (i mean the logic of where to get the config from and how to validate it)